libvast/include/vast/detail/generator.hpp

```diff
@@ -208,23 +208,16 @@ class [[nodiscard]] generator {
   generator(const generator& other) = delete;
 
   ~generator() {
-    if (m_coroutine)
+    if (m_coroutine) {
       m_coroutine.destroy();
+    }
   }
 
   generator& operator=(generator other) noexcept {
     swap(other);
     return *this;
   }
 
-  generator& operator=(generator&& other) noexcept {
-    if (m_coroutine)
-      m_coroutine.destroy();
-    m_coroutine = other.m_coroutine;
-    other.m_coroutine = nullptr;
-    return *this;
-  }
-
   iterator begin() {
```

Comment:
The commit introducing the move assignment was fixing a rule-of-five violation, which you seem to reintroduce by removing it. A better fix would probably to define the other version as `operator=(const generator&)`, which is how the assignment operator is supposed to look anyways.